### PR TITLE
Konstantin/appeals 59181

### PR DESCRIPTION
**Pull request by KonstantinShevtsov**
Resolves [APPEALS-59181](https://jira.devops.va.gov/browse/APPEALS-59181)

# Description
That PR adds 3 eligibility-related feature tests.

### CODE REVIEW

File: spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+require "rails_helper"
+
+RSpec.describe Api::Events::V1::DecisionReviewUpdatedController, type: :controller do
+  let!(:current_user) { User.authenticate! }
+  let(:api_key) { ApiKey.create!(consumer_name: "API TEST TOKEN") }
+  let!(:epe) { create(:end_product_establishment, :active_hlr, reference_id: 12_345_678) }
+  let(:review) { epe.source }
+
+  context "ineligible_to_eligible" do
```
Comment (jtsangVA) [inline spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:11]: please add more descriptive contexts. Include "request issues" e.g. ""ineligible_to_eligible request issues" so readers get more context.

File: spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+      context "updates issue" do
+        before do
+          request.headers["Authorization"] = "Token token=#{api_key.key_string}"
+        end
+
+        it "returns success response whith eligible request_issue" do
+          request.headers["Authorization"] = "Token token=#{api_key.key_string}"
+          post :decision_review_updated, params: valid_params
+          expect(existing_request_issue.reference_id).to eq("1234")
```
Comment (jtsangVA) [inline spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:113]: If these expects are for the prior state, then should be moved to above the POST line

File: spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+RSpec.describe Api::Events::V1::DecisionReviewUpdatedController, type: :controller do
+  let!(:current_user) { User.authenticate! }
+  let(:api_key) { ApiKey.create!(consumer_name: "API TEST TOKEN") }
+  let!(:epe) { create(:end_product_establishment, :active_hlr, reference_id: 12_345_678) }
+  let(:review) { epe.source }
+
+  context "ineligible_to_eligible" do
+    describe "POST #decision_review_updated" do
+      let!(:existing_request_issue) do
```
Comment (jtsangVA) [inline spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:13]: rename the issue variable to fit the context you're testing

File: spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+          expect(existing_request_issue.contention_removed_at).to eq(nil)
+          expect(existing_request_issue.contention_reference_id).to eq(123_456)
+        end
+      end
+    end
+  end
+
+  context "eligible_to_ineligible" do
+    describe "POST #decision_review_updated" do
+      let!(:existing_request_issue) do
```
Comment (jtsangVA) [inline spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:141]: rename here

File: spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+          expect(existing_request_issue.nonrating_issue_category).to eq("Military Retired Pay ELIGIBLE")
+          expect(existing_request_issue.nonrating_issue_description).to eq("UPDATED ELIGIBLE")
+          expect(existing_request_issue.closed_at).to eq("2023-12-07 20:49:05.000000000 -0500")
+          expect(existing_request_issue.closed_status).to eq("removed")
+        end
+      end
+    end
+  end
+
+  context "ineligible_to_ineligible" do
```
Comment (jtsangVA) [inline spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:261]: Is this just updating the description?

Comment (KonstantinShevtsov) [reply spec/feature/events/decision_review_updated/scenario_update_eligibility_spec.rb:261]: Yes

**Pull request merged**